python/tests/test_topology.py

```diff
@@ -45,43 +45,54 @@ def slice(
         record_provenance=True):
     """
     A clearer but slower implementation of TreeSequence.slice() defined in trees.py
+    expanded to slice at multiple intervals
     """
     if start is None:
-        start = 0
+        start = [0]
+    if np.ndim(start) == 0:
+        start = [start]
     if stop is None:
-        stop = ts.sequence_length
-
-    if start < 0 or stop <= start or stop > ts.sequence_length:
+        stop = [ts.sequence_length]
+    if np.ndim(stop) == 0:
+        stop = [stop]
+    zipRanges = list(zip(start, stop))
```

Comment:
Style nitpick: can we keep to snake_case for consistency please?

Reply:
sorry... old habits die hard

Reply:
No worries!